### ISSUE TITLE
Helm commands fail with truncate-string-ellipsis since todays build

### ISSUE THREAD

**Issue by SvenRieke** [emacs-helm/helm#2673]
### What happened?

During the compilation while installing the update:
```
Compiling file /home/sven/.emacs.d/elpa/helm-20240521.537/helm-x-files.el at Tue May 21 09:13:09 2024
helm-x-files.el:22:1:Error: Symbol’s function definition is void: truncate-string-ellipsis
```

Since then every helm command fails with this same error.

### How to reproduce?

Install the recent version on 
```
GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.16.0)
 of 2023-08-16, modified by Debian
```

### Helm Version

Melpa or NonGnuElpa

### Emacs Version

Emacs-27.1

### OS

GNU/Linux

### Relevant backtrace (if possible)

```text
Debugger entered--Lisp error: (void-function truncate-string-ellipsis)
  truncate-string-ellipsis()
  #f(compiled-function () #<bytecode 0x15909da35021>)()
  funcall(#f(compiled-function () #<bytecode 0x15909da35021>))
  eval((funcall #'#f(compiled-function () #<bytecode 0x15909da35021>)))
  custom-initialize-reset(helm-buffers-end-truncated-string (funcall #'#f(compiled-function () #<bytecode 0x15909da35021>)))
  custom-declare-variable(helm-buffers-end-truncated-string (funcall #'#f(compiled-function () #<bytecode 0x15909da35021>)) "The string to display at end of truncated buffer n..." :type string)
  byte-code("\301\302!\210\301\303!\210\301\304!\210\301\305!\210\301\306!\210\301\307!\210\301\310!\210\301\311!\210\301\312!\210\313\314\315\316\317\304%\210\320\321\322\323\324D..." [emacs-major-version require cl-lib mule-util helm helm-types helm-utils helm-grep helm-regexp helm-help helm-occur custom-declare-group helm-buffers nil "Buffers related Applications and libraries for Hel..." :group custom-declare-variable helm-boring-buffer-regexp-list funcall function #f(compiled-function () #<bytecode 0x15909da34f89>) "The regexp list that match boring buffers.\nBuffer ..." :type (repeat (choice regexp)) helm-white-buffer-regexp-list #f(compiled-function () #<bytecode 0x15909da34f95>) "The regexp list of not boring buffers.\nThese buffe..." (repeat (choice regexp)) helm-buffers-favorite-modes #f(compiled-function () #<bytecode 0x15909da34fa1>) "List of preferred mode to open new buffers with." (repeat (choice function)) helm-buffer-max-length #f(compiled-function () #<bytecode 0x15909da34fb1>) "Max length of buffer names before truncate.\nWhen d..." (choice (const :tag "Disabled" nil) (integer :tag "Length before truncate")) helm-buffer-details-flag #f(compiled-function () #<bytecode 0x15909da34fc1>) "Always show details in buffer list when non-nil." boolean helm-buffers-fuzzy-matching #f(compiled-function () #<bytecode 0x15909da34fd1>) "Fuzzy matching buffer names when non-nil.\nOnly buf..." helm-buffer-skip-remote-checking #f(compiled-function () #<bytecode 0x15909da34fe1>) "Ignore checking for `file-exists-p' on remote file..." helm-buffers-truncate-lines #f(compiled-function () #<bytecode 0x15909da34ff1>) "Truncate lines in `helm-buffers-list' when non-nil..." helm-buffers-left-margin-width ...] 10)
  require(helm-buffers)
  eval-buffer(#<buffer  *load*> nil "/home/sven/.emacs.d/elpa/helm-20240521.537/helm-fi..." nil t)  ; Reading at buffer position 973
  load-with-code-conversion("/home/sven/.emacs.d/elpa/helm-20240521.537/helm-fi..." "/home/sven/.emacs.d/elpa/helm-20240521.537/helm-fi..." nil t)
  autoload-do-load((autoload "helm-files" "Preconfigured `helm' for helm implementation of `f..." t nil) helm-find-files)
  command-execute(helm-find-files)
```

### Minimal configuration

- [X] I agree using a minimal configuration

**Comment by thierryvolpiatto** [comment]
SvenRieke ***@***.***> writes:

> What happened?
>
> During the compilation while installing the update:
>
> Compiling file /home/sven/.emacs.d/elpa/helm-20240521.537/helm-x-files.el at Tue May 21 09:13:09 2024
> helm-x-files.el:22:1:Error: Symbol’s function definition is void: truncate-string-ellipsis
>
> Since then every helm command fails with this same error.
>
> How to reproduce?
>
> Install the recent version on
>
> GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.16.0)
>  of 2023-08-16, modified by Debian

Indeed, `truncate-string-ellipsis` (the function) is not available in
emacs-27, I couldn't find it, even if the docstring says it is available
since 25.1:

,----[ truncate-string-ellipsis ]
| truncate-string-ellipsis is a byte-compiled Lisp function in
| `mule-util.el'.
| 
| (truncate-string-ellipsis)
| 
| Return the string used to indicate truncation.
| Use the value of the variable `truncate-string-ellipsis' when it's non-nil.
| Otherwise, return the Unicode character U+2026 "HORIZONTAL ELLIPSIS"
| when it's displayable on the selected frame, or `...'.  This function
| needs to be called on every use of `truncate-string-to-width' to
| decide whether the selected frame can display that Unicode character.
| 
|   Probably introduced at or before Emacs version 25.1.
`----

I inlined the definition (more or less) in the defcustom, so it should
be fixed now.

Thanks.

> Helm Version
>
> Melpa or NonGnuElpa
>
> Emacs Version
>
> Emacs-27.1
>
> OS
>
> GNU/Linux
>
> Relevant backtrace (if possible)
>
> Debugger entered--Lisp error: (void-function truncate-string-ellipsis)
>   truncate-string-ellipsis()
>   #f(compiled-function () #<bytecode 0x15909da35021>)()
>   funcall(#f(compiled-function () #<bytecode 0x15909da35021>))
>   eval((funcall #'#f(compiled-function () #<bytecode 0x15909da35021>)))
>   custom-initialize-reset(helm-buffers-end-truncated-string (funcall #'#f(compiled-function () #<bytecode 0x15909da35021>)))
>   custom-declare-variable(helm-buffers-end-truncated-string (funcall #'#f(compiled-function () #<bytecode 0x15909da35021>)) "The string to display at end of truncated buffer n..." :type string)
>   byte-code("\301\302!\210\301\303!\210\301\304!\210\301\305!\210\301\306!\210\301\307!\210\301\310!\210\301\311!\210\301\312!\210\313\314\315\316\317\304%\210\320\321\322\323\324D..." [emacs-major-version require cl-lib mule-util helm helm-types helm-utils helm-grep helm-regexp helm-help helm-occur custom-declare-group helm-buffers nil "Buffers related Applications and libraries for Hel..." :group custom-declare-variable helm-boring-buffer-regexp-list funcall function #f(compiled-function () #<bytecode 0x15909da34f89>) "The regexp list that match boring buffers.\nBuffer ..." :type (repeat (choice regexp)) helm-white-buffer-regexp-list #f(compiled-function () #<bytecode 0x15909da34f95>) "The regexp list of not boring buffers.\nThese buffe..." (repeat (choice regexp)) helm-buffers-favorite-modes #f(compiled-function () #<bytecode 0x15909da34fa1>) "List of preferred mode to open new buffers with." (repeat (choice function)) helm-buffer-max-length #f(compiled-function () #<bytecode 0x15909da34fb1>) "Max length of buffer names before truncate.\nWhen d..." (choice (const :tag "Disabled" nil) (integer :tag "Length before truncate")) helm-buffer-details-flag #f(compiled-function () #<bytecode 0x15909da34fc1>) "Always show details in buffer list when non-nil." boolean helm-buffers-fuzzy-matching #f(compiled-function () #<bytecode 0x15909da34fd1>) "Fuzzy matching buffer names when non-nil.\nOnly buf..." helm-buffer-skip-remote-checking #f(compiled-function () #<bytecode 0x15909da34fe1>) "Ignore checking for `file-exists-p' on remote file..." helm-buffers-truncate-lines #f(compiled-function () #<bytecode 0x15909da34ff1>) "Truncate lines in `helm-buffers-list' when non-nil..." helm-buffers-left-margin-width ...] 10)
>   require(helm-buffers)
>   eval-buffer(#<buffer  *load*> nil "/home/sven/.emacs.d/elpa/helm-20240521.537/helm-fi..." nil t)  ; Reading at buffer position 973
>   load-with-code-conversion("/home/sven/.emacs.d/elpa/helm-20240521.537/helm-fi..." "/home/sven/.emacs.d/elpa/helm-20240521.537/helm-fi..." nil t)
>   autoload-do-load((autoload "helm-files" "Preconfigured `helm' for helm implementation of `f..." t nil) helm-find-files)
>   command-execute(helm-find-files)
>
> Minimal configuration
>
> * I agree using a minimal configuration
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.

-- 
Thierry

**Comment by michael-heerdegen** [comment]
Thierry Volpiatto ***@***.***> writes:

> Indeed, `truncate-string-ellipsis` (the function) is not available in
> emacs-27, I couldn't find it, even if the docstring says it is available
> since 25.1:

The heuristic probably found the variable of the same name in NEWS.25.

Michael.

**Comment by SvenRieke** [comment]
I changed in helm-buffers.el temporarily this:

```
(defcustom helm-buffers-column-separator "…"
  "Separator for columns in buffer listing."
  :type 'string)
```
Now, everything works again.

**Comment by SvenRieke** [comment]
Fixed in helm-20240521.1457
